src/main/java/org/apache/lucene/analysis/kr/utils/DictionaryUtil.java

```diff
@@ -18,36 +18,127 @@
 
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Map;
 
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
 import org.apache.lucene.analysis.kr.morph.CompoundEntry;
 import org.apache.lucene.analysis.kr.morph.MorphException;
 import org.apache.lucene.analysis.kr.morph.WordEntry;
 
 public class DictionaryUtil {
 	
-	private static Trie<String,WordEntry> dictionary;
+	private static Supplier<Trie<String,WordEntry>> DICTIONARY = Suppliers.memoize(new Supplier<Trie<String, WordEntry>>() {
```

Comment:
I understand "keep the style as the surrounding file" but not having spaces between the type params in the generic declaration is crazy! ಠ_ಠ